Documentation/api-reference.md

```diff
@@ -0,0 +1,529 @@
+final DecisionScope decisionScope2 = new DecisionScope("myScope");
+
+final List<DecisionScope> decisionScopes = new ArrayList<>();
+decisionScopes.add(decisionScope1);
+decisionScopes.add(decisionScope2);
+
+Optimize.getPropositions(scopes, new AdobeCallbackWithError<Map<DecisionScope, Proposition>>() {
+    @Override
+    public void fail(final AdobeError adobeError) {
+        // handle error
+    }
+
+    @Override
+    public void call(Map<DecisionScope, Proposition> propositionsMap) {
+        if (propositionsMap != null && !propositionsMap.isEmpty()) {
+            // get the propositions for the given decision scopes
+            if (propositionsMap.contains(decisionScope1)) {
+                final Proposition proposition1 = propsMap.get(decisionScope1)
+                // read proposition1 offers
+            }
+            if (propositionsMap.contains(decisionScope2)) {
+                final Proposition proposition2 = propsMap.get(decisionScope2)
+                // read proposition2 offers
+            }
+        }
+    }
+});
+```
+
+## onPropositionsUpdate
+
+This API registers a permanent callback which is invoked whenever the Edge extension dispatches a response Event received from the Experience Edge Network upon a personalization query. The personalization query requests can be triggered by the `updatePropositions` API, Edge extension `sendEvent` API or launch consequence rules.
+
+### Java
+
+#### Syntax
+
+```java
+public static void onPropositionsUpdate(final AdobeCallback<Map<DecisionScope, Proposition>> callback)
+```
+
+* _callback_ `call` method is invoked with propositions map of type `Map<DecisionScope, Proposition>`. If the callback is an instance of `AdobeCallbackWithError`, and if the operation times out or an error occurs in retrieving propositions, the `fail` method is invoked with the appropriate `AdobeError`.
+
+#### Example
+
+```java
+Optimize.onPropositionsUpdate(new AdobeCallbackWithError<Map<DecisionScope, Proposition>>() {
+    @Override
+    public void fail(final AdobeError adobeError) {
+        // handle error
+    }
+
+    @Override
+    public void call(final Map<DecisionScope, Proposition> propositionsMap) {
+        if (propositionsMap != null && !propositionsMap.isEmpty()) {
+            // handle propositions
+        }
+    }
+});
+```
+
+## registerExtension
+
+This API can be invoked to register the Optimize extension with the Mobile Core.
+
+### Java
+
+#### Syntax
+
+```java
+public static void registerExtension()
+```
+
+#### Example
+
+```java
+Optimize.registerExtension();
+```
+
+## resetIdentities
+
+This `MobileCore` API can also be invoked to clear out the client-side data for the Optimize extension, such as the in-memory propositions cache.
+For details on syntax, usage and availability, refer to [Mobile Core - Reset identities](https://github.com/adobe/aepsdk-core-android).
```

Comment:
Recommend slight modification here:

This MobileCore API is a request to each extension to reset its identities. Every extension responds to this request in its own unique manner. For example, Optimize extension uses this API call to clear out its client-side in-memory propositions cache. For more details, see [MobileCore - API reference](https://github.com/adobe/aepsdk-core-android/blob/main/Documentation/MobileCore/api-reference.md).

[!WARNING] This API call can lead to unintended SDK behavior, e.g. resetting of Experience Cloud ID (ECID). So it should be sparingly used and extreme caution should be followed!